homeassistant/components/geniushub/water_heater.py

```diff
@@ -0,0 +1,141 @@
+GH_STATE_TO_HA = {
+    'off': STATE_OFF,
+    'timer': STATE_AUTO,
+    'footprint': None,
+    'away': None,
+    'override': STATE_MANUAL,
+    'early': None,
+    'test': None,
+    'linked': None,
+    'other': None,
+}
+
+GH_DEVICE_STATE_ATTRS = ['type', 'override']
+
+
+async def async_setup_platform(hass, hass_config, async_add_entities,
+                               discovery_info=None):
+    """Set up the Genius Hub water_heater entities."""
+    client = hass.data[DOMAIN]['client']
+
+    entities = [GeniusWaterHeater(client, z)
+                for z in client.hub.zone_objs if z.type in GH_WATER_HEATERS]
+
+    async_add_entities(entities)
+
+
+class GeniusWaterHeater(WaterHeaterDevice):
+    """Representation of a Genius Hub water_heater device."""
+
+    def __init__(self, client, zone):
+        """Initialize the water_heater device."""
+        self._client = client
+        self._objref = zone
+        self._id = zone.id
+        self._name = zone.name
+
+        self._operation_list = list(HA_OPMODE_TO_GH)
+
+    @property
+    def name(self):
+        """Return the name of the water_heater device."""
+        return self._objref.name
+
+    @property
+    def device_state_attributes(self):
+        """Return the device state attributes."""
+        tmp = self._objref.__dict__.items()
+        state = {k: v for k, v in tmp if k in GH_DEVICE_STATE_ATTRS}
+
+        return {'status': state}
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        return self._objref.temperature
+
+    @property
+    def target_temperature(self):
+        """Return the temperature we try to reach."""
+        return self._objref.setpoint
+
+    @property
+    def min_temp(self):
+        """Return max valid temperature that can be set."""
+        return GENIUSHUB_MIN_TEMP
+
+    @property
+    def max_temp(self):
+        """Return max valid temperature that can be set."""
+        return GENIUSHUB_MAX_TEMP
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement."""
+        return TEMP_CELSIUS
+
+    @property
+    def supported_features(self):
+        """Return the list of supported features."""
+        return GENIUSHUB_SUPPORT_FLAGS
+
+    @property
+    def operation_list(self):
+        """Return the list of available operation modes."""
+        return self._operation_list
+
+    @property
+    def current_operation(self):
+        """Return the current operation mode."""
+        return GH_STATE_TO_HA.get(self._objref.mode)
+
+    async def async_set_operation_mode(self, operation_mode):
+        """Set a new operation mode for this zone."""
+        await self._objref.set_mode(HA_OPMODE_TO_GH.get(operation_mode))
+
+    async def async_set_temperature(self, **kwargs):
+        """Set a new target temperature for this zone."""
+        temperature = kwargs.get(ATTR_TEMPERATURE)
+        await self._objref.set_override(temperature, 3600)  # 1 hour
```

Comment:
Is it ok to pass `None` as argument to `_objref.set_override`?

Reply:
Are you saying that `kwargs.get(ATTR_TEMPERATURE)` could `== None`?

Anyway, yes:
```python
async def set_override(self, setpoint=None, duration=3600):
```

Reply:
Sorry, I was mistaken. I was thinking about the climate component temp set schema. There, `ATTR_TEMPERATURE` is optional. It's not optional here.